### PR TITLE
[filter] Extract inference wrapper header @open sesame 02/19 17:42

### PR DESCRIPTION
**Pull request by zhoonit**
This patch extracts inference wrapper header for testability and better
maintainability.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.hh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+/**
+ * @file	tensor_filter_nntrainer.hh
+ * @date	09 Sept 2020
+ * @brief	nntrainer inference module for tensor_filter gstreamer plugin header
+ * @note  The clas has been exposed from tensor_filter_nntrainer.cc to unittest
+ * @see		http://github.com/nnstreamer/nnstreamer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ * This is the per-NN-framework plugin (tensorflow-lite) for tensor_filter.
```
Comment (kparichay) [inline nnstreamer/tensor_filter/tensor_filter_nntrainer.hh:15]: tensorflow-lite?

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.hh [hunk 2]
```diff
@@ -0,0 +1,106 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/**
+ * GStreamer Tensor_Filter, nntrainer Module
```
Comment (kparichay) [inline nnstreamer/tensor_filter/tensor_filter_nntrainer.hh:3]: GStreamer -> NNStreamer

File: nnstreamer/tensor_filter/meson.build [hunk 3]
```diff
@@ -22,15 +22,15 @@ filter_subplugin_install_dir = subplugin_install_prefix / 'filters'
 shared_library('nnstreamer_filter_nntrainer',
   nnstreamer_filter_nntrainer_sources,
   dependencies: nnstreamer_filter_nntrainer_deps,
-  include_directories: nntrainer_inc,
+  include_directories: [nntrainer_inc, '.'],
```
Comment (kparichay) [inline nnstreamer/tensor_filter/meson.build:25]: please add a comment here to not install this header file.

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.cc [hunk 4]
```diff
@@ -7,7 +7,7 @@
  * @file	tensor_filter_nntrainer.cc
  * @date	09 Sept 2020
  * @brief	nntrainer inference module for tensor_filter gstreamer plugin
- * @see		http://github.com/nnsuite/nnstreamer
+ * @see		http://github.com/nnstreamer/nnstreamer
```
Comment (dongju-chae) [inline nnstreamer/tensor_filter/tensor_filter_nntrainer.cc:10]: http://github.com/nnstreamer/nntrainer?

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.hh (hunk 1 above)
Comment (zhoonit) [reply nnstreamer/tensor_filter/tensor_filter_nntrainer.hh:15]: I think the original intention is `e.g) tensorflow-lite`, I will update.

File: nnstreamer/tensor_filter/meson.build (hunk 3 above)
Comment (zhoonit) [reply nnstreamer/tensor_filter/meson.build:25]: Done

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.hh (hunk 2 above)
Comment (zhoonit) [reply nnstreamer/tensor_filter/tensor_filter_nntrainer.hh:3]: Done

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.hh (hunk 1 above)
Comment (zhoonit) [reply nnstreamer/tensor_filter/tensor_filter_nntrainer.hh:15]: Done

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**